fetch_tags.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+###
+# EXIT CODES
+# 0 - success
+# 1 - parameter error
+# 50 - dockerhub pull error
+# 51 - no valid docker image tag
+# 52 - no old tag file found
+# 53 - max tag count iteration reached
+# 100 - new tag found, pr_script triggered
+###
+
+# Sample envionment variables
+# IMAGE_ID="weaver-worker"
+# DOCKERHUB_REPO="pavics/weaver"
+# TAG_FILTER="^[0-9]+(\\.[0-9]+)*$"
+# BUMP_TAG="WEAVER_WORKER_IMAGE"
+# BUMP_FILE="birdhouse/default.env"
+
+REQUIRED_ENV_VARS='
+    IMAGE_ID
+    DOCKERHUB_REPO
+    TAG_FILTER
+    BUMP_TAG
+    BUMP_FILE
+'
+
+# args parsing
+for env_var in $REQUIRED_ENV_VARS
+do
+    if [[ ! -v "${env_var}" ]]; then
+        echo "[ERROR] [$0] Missing ${env_var} environment variable. Exiting."
+        exit 1
+    fi
+done
+
+echo "[STEP] [$0] [$IMAGE_ID] Fetch tags"
+
+DATA_DIR="data"
+
+
+# get latest image tag from dockerhub
+mkdir -p data
+NAME=${DOCKERHUB_REPO//\//_}_$IMAGE_ID
+NEW_FILENAME=$NAME.new
+NEW_FILEPATH=$DATA_DIR/$NEW_FILENAME
+DOCKERHUB_IMAGE_TAGS=$(wget -q $DOCKERHUB_IMAGE_TAGS_URL$DOCKERHUB_REPO/tags?page_size=1024 -O-)
+MAX_TAG_ITERATION=$(echo $DOCKERHUB_IMAGE_TAGS | jq '.count')
+TAG_INDEX=0
+TAG_FILTER=${TAG_FILTER//\\\\/\\}    # replace \\ with \ in regex
+
+while [[ ! $LATEST_TAG =~ $TAG_FILTER ]]
+do
+    # assumption that results are always in the same order, most recent first
+    LATEST_TAG=$(echo $DOCKERHUB_IMAGE_TAGS | jq '.results['$TAG_INDEX'].name' | tr -d \")
+    TAG_INDEX=$TAG_INDEX+1
+
+    if [[ "$TAG_INDEX" -eq "$MAX_TAG_ITERATION" ]]; then
+        echo "[WARNING] [$0] Reached max tag iteration count. Exiting."
+        exit 53
+    fi
+done
+
+if [ "$LATEST_TAG" == "" ]; then
+    echo "[INFO] [$0] [$IMAGE_ID] No valid tag found. Exiting."
+    exit 51
+fi
+
+echo $LATEST_TAG > $NEW_FILEPATH
+
+
+# make sure data is persisted to file
+if [ ! -f "$NEW_FILEPATH" ]; then
+    echo "[WARNING] [$0] [$IMAGE_ID] Error when pulling DockerHub data. Exiting."
+    exit 50
+fi
+
+
+# diff with old file
+OLD_FILENAME=$NAME.old
+OLD_FILEPATH=$DATA_DIR/$OLD_FILENAME
+
+NEW_TAG_FOUND=false
+
+if [ -f "$OLD_FILEPATH" ] && [ -f "$NEW_FILEPATH" ]; then
+    DIFF=$(diff $OLD_FILEPATH $NEW_FILEPATH)
+
+    if [ "$DIFF"  == "" ]; then
```

Comment:
`if diff $OLD_FILEPATH $NEW_FILEPATH; then` diff exit 0 when no diff found.  Can just use its exit code than parse the diff output.